corehq/apps/case_search/models.py

```diff
@@ -35,26 +32,41 @@
 
 @attr.s(frozen=True)
 class CaseSearchRequestConfig:
+    criteria = attr.ib(kw_only=True)
+    case_type = attr.ib(kw_only=True, default=None)
     data_registry = attr.ib(kw_only=True, default=None)
     expand_id_property = attr.ib(kw_only=True, default=None)
 
+    @property
+    def case_types(self):
+        return self.case_type if isinstance(self.case_type, list) else [self.case_type]
```

Comment:
Does this mean that `self.case_type` can either be a list or a string? What if you named property above to `case_types` and always required that to be a list? Otherwise this feels a bit confusing.

Reply:
the request parameter is called `case_type` which is why the attribute has that name. The fact that it can be a string or a list is because that's how the pre-processing is done one L57:
```
params = {k: v[0] if v and len(v) == 1 else v for k, v in request_dict.lists()}
```

We could do treat this different for each param but that adds complexity as well. I'm open to ideas though.

Reply:
Up to you, I think. It seems like you already do some manipulation of different params on L68, so you could follow the same pattern to keep this class cleaner?

Reply:
@proteusvacuum I've made some changes to clean things up a bit. Please take a look.